docs/src/main/asciidoc/building-my-first-extension.adoc

```diff
@@ -383,6 +389,103 @@ The key points are:
 <3> We add  the `quarkus-bootstrap-maven-plugin` to generate the Quarkus extension descriptor included into the runtime artifact which links it with the corresponding deployment artifact.
 <4> We add  the `quarkus-extension-processor` to the compiler annotation processors.
 
+=== Gradle setup
+
+Quarkus does not provide any way to initialize a Gradle project for extension yet.
+
+As mentionned before, an extension is composed of two modules:
+
+* `runtime`
+* `deployment`
+
+We are going to create a Gradle multi module project with those two modules. Here is a simple `settings.gradle` example file:
+
+[source, groovy]
+----
+pluginManagement {
+    repositories {
+        mavenCentral()
+        gradlePluginPortal()
+    }
+    plugins {
+        id 'io.quarkus.extension' version "${quarkus.version}" <1>
+    }
+}
+
+include 'runtime', 'deployment' <2>
+
+rootProject.name = 'greeting-extension'
+----
+
+<1> Configure the quarkus extension plugin version
+<2> Include both `runtime` and `deployment` modules
+
+Here is a sample of a root `build.gradle` file:
+
+[source, groovy]
+----
+subprojects {
+    apply plugin: 'java-library' <1>
+    apply plugin: 'maven-publish' <2>
+
+    group 'org.acme' <3>
+    version '1.0-SNAPSHOT'
+}
+----
+
+<1> Apply the `java-library` plugin for all sub-modules
+<2> Apply the `maven-publish` plugin used to publish our artifacts
+<3> Globally set the group id used for publication
+
+The `io.quarkus.extension` plugin will be used in order to help us building the extension.
+The plugin will *only* be applied to the `runtime` module.
+
+==== The deployment module
+
+The deployment module does not require any specific plugin.
+Here is an example of a minimal `build.gradle` file for the `deployment` module:
+
+[source, groovy]
+----
+name = 'greeting-extension-deployment' <1>
+
+dependencies {
+    implementation project(':runtime') <2>
+
+    implementation platform("io.quarkus:quarkus-bom:${quarkus.version}")
+
+    testImplementation 'io.quarkus:quarkus-junit5-internal'
+}
+----
+
+<1> By convention, the deployment module has the `-deployment` suffix (`greeting-extension-deployment`).
+<2> The deployment module *must* depend on the `runtime` module.
+
+==== The runtime module
+
+The runtime module applies the `io.quarkus.extension` plugin. This will:
+
+* Add `quarkus-extension-process` as annotation processor to both modules.
+* Generate extension description files.
+
+Here is an example of `build.gradle` file for the `runtime` module:
+
+[source, groovy]
+----
+plugins {
+    id 'io.quarkus.extension' <1>
+}
+
+name = 'greeting-extension' <2>
```

Comment:
Is there a way to set the description of the extension too?

Reply:
yes sure I'm adding one.